RequestService/RequestService.Repo/Repository.cs

```diff
@@ -1703,5 +1703,50 @@ public async Task UpdateJobsToCancelledFromNewOrOpen()
                 }
             }
         }
+
+        public List<RequestSummary> GetRequestsByFilter(GetRequestsByFilterRequest request, List<int> referringGroups)
+        {
+            IQueryable<Request> requests = _context.Request
+               .Include(i => i.Shift)
+               .Include(i => i.Job)
+               .ThenInclude(i => i.JobAvailableToGroup)
+               .Include(i => i.Job)
+               .ThenInclude(i => i.RequestJobStatus);
+               
+
+            if (requests == null || requests.Count() == 0)
+            {
+                return new List<RequestSummary>();
+            }
+
+            if (referringGroups.Count > 0)
+            {
+                requests = requests.Where(x => referringGroups.Contains(x.ReferringGroupId));
+            }
+
+            if (request.Groups?.Groups.Count > 0)
+            {
+                requests = requests.Where(x => x.Job.SelectMany(x => x.JobAvailableToGroup).Any(a => request.Groups.Groups.Contains(a.GroupId)));
+            }
+
+            if (request.DateFrom.HasValue)
+            {
+                requests = requests.Where(x => x.Shift.StartDate.AddMinutes(x.Shift.ShiftLength) >= request.DateFrom.Value);
```

Comment:
At some point we should allow Tasks as well as Shifts to be filtered on the basis of DateFrom and DateTo.